File: IOS/Sources/OpenRoaming/OpenRoaming/Services/SetPrivacySettingsService.swift
//
//  SetPrivacySettingsService.swift
//  OpenRoaming
//
//  Created by Luis Vasquez on 06/05/20.
//  Copyright © 2020 Cisco. All rights reserved.
//

import UIKit

class SetPrivacySettingsService: Service {
    
    var acceptPrivacySettings:Bool?
    
    /**
    * Not implemented.
    * Use start(context: Context, acceptPrivacySettings: Boolean, serviceHandler: ServiceHandler).
    */
    override func start(serviceHandler: @escaping ServiceHandler) {
        fatalError("Not implemented")
    }
    
    func start(acceptPrivacySettings: Bool, serviceHandler: @escaping ServiceHandler) {
        self.acceptPrivacySettings = acceptPrivacySettings
        super.start(serviceHandler: serviceHandler)
    }
    
    override func onStart(headerParams: Array<HeaderParam>, serviceHandler: @escaping ServiceHandler) {
        let urlString = Urls.privacySettingsUrl() + "/" + String(acceptPrivacySettings!)
        let requestData = RequestData(url: urlString, headerParams: headerParams, method: HTTPMethod.PUT)
        call(requestData: requestData, serviceHandler: serviceHandler)
    }
}

File: Android/src/main/java/com/cisco/or/sdk/services/SetPrivacySettingsService.kt
package com.cisco.or.sdk.services

import android.content.Context
import com.cisco.or.sdk.types.ServiceHandler
import com.cisco.or.sdk.utils.Urls
import com.cisco.or.sdk.utils.Utils

/**
 * Accepts or not to share user e-mail
 */
internal object SetPrivacySettingsService : Service(){
    private var acceptPrivacySettings: Boolean = false

    /**
     * Not implemented.
     * Use start(context: Context, acceptPrivacySettings: Boolean, serviceHandler: ServiceHandler).
     */
    override fun start(context: Context, serviceHandler: ServiceHandler) {
        if(Utils.internalCall(Service::class.qualifiedName.toString(), "start")){
            super.start(context, serviceHandler)
        }else{
            throw NotImplementedError()
        }
    }

    /**
     * Sends the accept share email answer.
     * @param context Context of the functionality
     * @param acceptPrivacySettings True = accept privacy settings
     * @param serviceHandler Type of handler to response
     */
    fun start(context: Context, acceptPrivacySettings: Boolean, serviceHandler: ServiceHandler) {
        this.acceptPrivacySettings = acceptPrivacySettings
        super.start(context,serviceHandler)
    }

    override fun onStart(context: Context, headerParams: ArrayList<HeaderParam>, serviceHandler: ServiceHandler) {
        val url = Urls.privacySettingsUrl() + "/" + acceptPrivacySettings
        val requestData = RequestData(url, headerParams, HTTP_METHOD.PUT, null)
        call(context, requestData, RESPONSE_FORMATTER.TEXT, serviceHandler)
    }
}